master/internal/rm/kubernetesrm/resource_pool.go

```diff
@@ -269,6 +269,10 @@ func (k *kubernetesResourcePool) ValidateResources(
 	k.reschedule = true
 
 	fulfillable := k.maxSlotsPerPod >= msg.Slots
+
+	if !msg.IsSingleNode {
```

Comment:
nit: it'd be nice to have this return specific errors rather than the caller trying to describe what could possibly go wrong in this function.

and then assignResources could reuse the error defs, too.

Reply:
yeah, i thought about this but didn't want to get too carried away with refactors. anyway, good call, done.